sdk/core/azure-core/azure/core/pipeline/transport/_base.py

```diff
@@ -220,18 +220,31 @@ class HttpRequest(object):
     :param str url: At least complete scheme/host/path
     :param dict[str,str] headers: HTTP headers
     :param files: Files list.
-    :param data: Body to be sent.
+    :param data: Body to be sent. If you want to send a json body, you should use the
+     `json` kwarg instead. We will handle json serialization for your data.
     :type data: bytes or str.
+    :keyword json: A JSON serializable object. Serializes your inputted object. Use this
+     instead of data if you wish for us to handle json serialization of your object for you.
+    :raises: TypeError if you input a non json serializable object through kwarg `json`
     """
 
-    def __init__(self, method, url, headers=None, files=None, data=None):
-        # type: (str, str, Mapping[str, str], Any, Any) -> None
+    def __init__(self, method, url, headers=None, files=None, data=None, **kwargs):
+        # type: (str, str, Mapping[str, str], Any, Any, Any) -> None
         self.method = method
         self.url = url
         self.headers = _case_insensitive_dict(headers)
         self.files = files
-        self.data = data
         self.multipart_mixed_info = None  # type: Optional[Tuple]
+        json_kwarg = "json" in kwargs
+        body_input = [files, data]
+        if json_kwarg and len([b for b in body_input if b]) > 0:
```

Comment:
Ideally I would throw a `ValueError` if more than one of `files`, `data`, or `json` is specified. However, I fear throwing if `files` and `data` are specified would be a breaking change, cc @johanste for confirmation

Reply:
Let's start with a `warnings.warn`.